docsrc/hello_world.rst

```diff
@@ -61,88 +54,90 @@ By default, the Stan program is compiled on instantiation.
     print(bernoulli_model)
 
             
-Run the HMC-NUTS sampler
-^^^^^^^^^^^^^^^^^^^^^^^^
+Data inputs
+^^^^^^^^^^^
+
+CmdStanPy accepts input data either as a Python dictionary which maps data variable names
+to values, or as the corresponding JSON file.
+
+The bernoulli model requires two inputs: the number of observations `N`, and
+an N-length vector `y` of binary outcomes.
+The data file `bernoulli.data.json` contains the following inputs:
+
+.. code::
+
+   {
+    "N" : 10,
+    "y" : [0,1,0,0,0,0,0,0,0,1]
+   }
+
+
 
-The :ref:`class_cmdstanmodel` method ``sample`` is used to do Bayesian inference
+Fitting the model
+^^^^^^^^^^^^^^^^^
+
+The :meth:`cmdstanpy.CmdStanModel.sample` method is used to do Bayesian inference
 over the model conditioned on data using  using Hamiltonian Monte Carlo
 (HMC) sampling. It runs Stan's HMC-NUTS sampler on the model and data and
 returns a :ref:`class_cmdstanmcmc` object.  The data can be specified
-either as a filepath or a Python dict; in this example, we use the
+either as a filepath or a Python dictionary; in this example, we use the
 example datafile `bernoulli.data.json`:
 
+By default, the ``sample`` method runs 4 sampler chains.
+The ``output_dir`` argument is an optional argument which specifies
+the path to the output directory used by CmdStan.
+If this argument is omitted, the output files are written
+to a temporary directory which is deleted when the current Python session is terminated. [#]_
 
-By default, the ``sample`` command runs 4 sampler chains.
-This is a set of per-chain 
-`Stan CSV files <https://mc-stan.org/docs/cmdstan-guide/stan-csv.html#mcmc-sampler-csv-output>`__
-The filenames follow the template '<model_name>-<YYYYMMDDHHMM>-<chain_id>'
-plus the file suffix '.csv'.
-There is also a correspondingly named file with suffix '.txt'
-which contains all messages written to the console.
-If the ``output_dir`` argument is omitted, the output files are written
-to a temporary directory which is deleted when the current Python session is terminated.
 
 .. code-block:: python
```

Comment:
I think it would be really valuable to replace these with IPython directives so that the code is actually run and output is displayed. I'm not sure if we can get cmdstan to run on readthedocs though - do we build the docs ahead of time, or do they need to be able to do it? Determines if this is possible

Reply:
Actually, we could probably fake it with (hidden) calls to fromcsv if we can't get cmdstan on rtd. I think it would be very valuable to actually see what all these print statements are doing